### PR TITLE
feat(auth): fetch ecosystem auth options from server

### PR DESCRIPTION
**Pull request by gregfromstl**
## PR-Codex overview
The focus of this PR is to refactor ecosystem wallet authentication options handling and improve UI consistency in the wallet components.

### Detailed summary
- Refactored handling of ecosystem wallet authentication options
- Updated UI components for better consistency and readability
- Added loading screen for ecosystem wallet authentication options retrieval
- Improved code structure and organization

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by gregfromstl** [comment]
* **#3528** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/3528?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @gregfromstl and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/thirdweb/src/react/web/wallets/in-app/InAppWalletFormUI.tsx [hunk 1]
```diff
@@ -79,8 +79,10 @@ export function InAppWalletFormUIScreen(props: InAppWalletFormUIProps) {
           />
           <Spacer y="lg" />
         </>
-      ) : (
+      ) : isCompact ? (
         <ModalHeader onBack={onBack} title={props.inAppWalletLocale.signIn} />
+      ) : (
+        <></>
```
Comment (jnsdls) [inline packages/thirdweb/src/react/web/wallets/in-app/InAppWalletFormUI.tsx:85]: can't this just be `null`?

File: packages/thirdweb/src/react/web/wallets/shared/ConnectWalletSocialOptions.tsx [hunk 2]
```diff
@@ -87,27 +90,35 @@ export const ConnectWalletSocialOptions = (
...
+    queryKey: ["auth-options", wallet.id],
+    queryFn: async () => {
+      if (isEcosystemWallet(wallet)) {
+        return (
+          (await getEcosystemWalletAuthOptions(wallet.id)) ?? defaultAuthOptions
+        );
+      }
+      return wallet.getConfig()?.auth?.options ?? defaultAuthOptions;
+    },
+    staleTime: 0,
```
Comment (jnsdls) [inline packages/thirdweb/src/react/web/wallets/shared/ConnectWalletSocialOptions.tsx:103]: this is the default, if you meant to fetch this once and never again (per session) use: `Infinity` instead

File: packages/thirdweb/src/wallets/ecosystem/get-ecosystem-wallet-auth-options.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  walletId: EcosystemWalletId,
+): Promise<AuthOption[] | undefined> {
+  const headers = new Headers();
+  headers.set("x-ecosystem-id", walletId);
+  const res = await fetch(
+    `${getThirdwebBaseUrl("inAppWallet")}/api/2024-05-05/ecosystem-wallet`,
+    {
+      headers,
+    },
+  );
```
Comment (jnsdls) [inline packages/thirdweb/src/wallets/ecosystem/get-ecosystem-wallet-auth-options.ts:21]: you could instead set the header directly in the init options for "headers"

```suggestion
  const res = await fetch(
    `${getThirdwebBaseUrl("inAppWallet")}/api/2024-05-05/ecosystem-wallet`,
    {
      headers: {
        "x-ecosystem-id": walletId
      },
    },
  );
```

**Review by jnsdls — CHANGES_REQUESTED** [review]
couple of things

**Review by jnsdls — APPROVED** [review]

**Pull request merged**